### PR TITLE
EVG-14623: PatchTasks query logging

### PR DESCRIPTION
**Pull request by SupaJoon**
https://jira.mongodb.org/browse/EVG-14623

### CODE REVIEW

File: graphql/resolvers.go [hunk 1]
```diff
@@ -2452,28 +2452,77 @@ func (r *taskResolver) BaseTask(ctx context.Context, obj *restModel.APITask) (*r
...
+	ticket := "EVG-14623"
+	grip.Warning(message.WrapError(err, message.Fields{
+				"message":           "1",
+				"ticket":           ticket,
+				"task": 			*obj.Id,
+			}))
 	if err != nil {
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting service model for APITask %s: %s", *obj.Id, err.Error()))
 	}
+	grip.Warning(message.WrapError(err, message.Fields{
```
Comment (ablack12) [inline graphql/resolvers.go:2464]: This log doesn't tell you anything really I think

File: graphql/resolvers.go [hunk 2]
```diff
@@ -2452,28 +2452,77 @@ func (r *taskResolver) BaseTask(ctx context.Context, obj *restModel.APITask) (*r
...
+				"message":           "2",
+				"ticket":           ticket,
+				"task": 			*obj.Id,
+			}))
 	t, ok := i.(*task.Task)
+	grip.Warning(message.WrapError(err, message.Fields{
```
Comment (ablack12) [inline graphql/resolvers.go:2470]: If error is nil, WrapError won't log anything, so in places where you've already validated err != nil you'll want a different if condition.

File: graphql/resolvers.go [hunk 3]
```diff
@@ -2452,28 +2452,77 @@ func (r *taskResolver) BaseTask(ctx context.Context, obj *restModel.APITask) (*r
...
 	for _, execTaskID := range t.ExecutionTasks {
 		execT, err := task.FindByIdExecution(execTaskID, &t.Execution)
+		grip.Warning(message.WrapError(err, message.Fields{
+			"message":           "6",
+			"ticket":           ticket,
+			"task": 			*obj.Id,
+			"execTask": execTaskID,
+		}))
 		if err != nil || execT == nil {
 			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error while getting execution task with id: %s : %s", execTaskID, err.Error()))
```
Comment (ablack12) [inline graphql/resolvers.go:2501]: This is a potential culprit: If err == nil and execT == nil, then err.Error() is a nil pointer reference, which is why we always have two if checks for database lookups

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: graphql/resolvers.go (hunk 3 above)
Comment (SupaJoon) [reply graphql/resolvers.go:2501]: WOW thank you!!!!

File: graphql/resolvers.go (hunk 1 above)
Comment (SupaJoon) [reply graphql/resolvers.go]: ah, makes sense. I should probably not be using WrapError

**Comment by SupaJoon** [comment]
I updated my code changes so we log information about the empty execution task instead of fail the entire resolver. This way, spruce users can at least view a subset of information instead of an error banner and we can understand the bug more closely from the logs.

File: graphql/resolvers.go [hunk 4]
```diff
@@ -2465,15 +2465,24 @@ func (r *taskResolver) ExecutionTasksFull(ctx context.Context, obj *restModel.AP
...
-		if err != nil {
-			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Unable to convert task: %s to APITask", execT.Id))
+		if execT != nil {
+			apiTask := &restModel.APITask{}
+			if err = apiTask.BuildFromService(execT); err != nil {
+				return nil, InternalServerError.Send(ctx, fmt.Sprintf("Unable to convert task: %s to APITask", execT.Id))
+			}
+			executionTasks = append(executionTasks, apiTask)
+		} else {
+			grip.Warning(message.Fields{
```
Comment (ablack12) [inline graphql/resolvers.go:2478]: Will not erroring here cause weird behavior for users? (like execution tasks are missing but it won't be obvious to them why)

**Review by ablack12 — CHANGES_REQUESTED** [review]
Sorry I didn't re-review until now! Definitely feel free to ping me if it's been like 24 hours since you've heard back

File: graphql/resolvers.go (hunk 4 above)
Comment (ablack12) [reply graphql/resolvers.go:2478]: This is the kind of bug that would get lost in splunk for us unless someone asked us what had happened.

Comment (SupaJoon) [reply graphql/resolvers.go:2478]: Right, these code changes will allow the task table to render without the offending execution tasks and will log the task id's that were not found in the DB. I'm not really sure why the execution tasks weren't found but I hope a list of missing task id's can help us identify a similarity amongst the set to help us debug an issue in the task lifecycle or somewhere else.

**Review by ablack12 — APPROVED** [review]

**Pull request merged**